readthedocs/doc_builder/python_environments.py

```diff
@@ -72,6 +74,120 @@ def delete_existing_venv_dir(self):
             )
             shutil.rmtree(venv_dir)
 
+    def install_build_tools(self):
+        build_os = getattr(self.config.build, 'os', None)
+        build_tools = getattr(self.config.build, 'tools', None)
+        if not build_os or not build_tools:
+            log.debug(
+                'Not installing "build.tools" because they are not defined. '
+                'build.os=%s build.tools=%s',
+                build_os,
+                build_tools,
+            )
+            return
+
+        if settings.RTD_DOCKER_COMPOSE:
+            # Create a symlink for ``root`` user to use the same ``.asdf``
+            # installation than ``docs`` user. Required for local building
+            # since everything is run as ``root`` when using Local Development
+            # instance
+            cmd = [
+                'ln',
+                '-s',
+                '/home/docs/.asdf',
+                '/root/.asdf',
+            ]
+            self.build_env.run(
+                *cmd,
+            )
+
+        for tool, version in build_tools.items():
+            # TODO: ask config file object for the specific tool version required by asdf
+            version = settings.RTD_DOCKER_BUILD_SETTINGS['tools'][tool][version]
```

Comment:
you can use 

```suggestion
            version = config.settings['tools'][tool][version]
```

Not sure if a method is needed, could have an object for each entry like `config.build.tools.python.version/config.build.tools.python.full_version` or something like that.

Reply:
`config.build.tools.python.full_version` works to me 👍🏼

Reply:
I'll use `config.settings` in the meanwhile.

Reply:
Hrm, why not using the property we already have as `config.python_full_version`? https://github.com/readthedocs/readthedocs.org/pull/8478/files/3762ca7bdec580653c0acccc27568917e4b34406..2d57be8e018283a16c4b51bd65e116e459c17dbd#diff-3fd49db79e50439b527e4310471c55d790ebc32ee55ffe1f12f83b9b6d5069dcR284

Reply:
> why not using the property we already have as config.python_full_version?

That will solve the problem for python only, go, rust, node will be using a different method.